source/extensions/filters/http/common/pass_through_filter.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include "envoy/http/filter.h"
+
+namespace Envoy {
+namespace Http {
+
+// A decoder filter which passes all data through with Continue status.
+class PassThroughDecoderFilter : public StreamDecoderFilter {
+  // Http::StreamFilterBase
+  void onDestroy() override {}
+
+  // Http::StreamDecoderFilter
+  Http::FilterHeadersStatus decodeHeaders(Http::HeaderMap&, bool) override {
+    return Http::FilterHeadersStatus::Continue;
+  }
+  Http::FilterDataStatus decodeData(Buffer::Instance&, bool) override {
+    return Http::FilterDataStatus::Continue;
+  }
+
+  Http::FilterTrailersStatus decodeTrailers(Http::HeaderMap&) override {
+    return Http::FilterTrailersStatus::Continue;
+  }
+  void setDecoderFilterCallbacks(Http::StreamDecoderFilterCallbacks& callbacks) override {
+    decoder_callbacks_ = &callbacks;
+  }
+
+protected:
+  Http::StreamDecoderFilterCallbacks* decoder_callbacks_{};
+};
+
+// An encoder filter which passes all data through with Continue status.
+class PassThroughEncoderFilter : public StreamEncoderFilter {
+public:
+  // Http::StreamFilterBase
+  void onDestroy() override {}
+
+  // Http::StreamEncoderFilter
+  Http::FilterHeadersStatus encode100ContinueHeaders(Http::HeaderMap&) override {
+    return Http::FilterHeadersStatus::Continue;
+  }
+  Http::FilterHeadersStatus encodeHeaders(Http::HeaderMap&, bool) override {
+    return Http::FilterHeadersStatus::Continue;
+  }
+  Http::FilterDataStatus encodeData(Buffer::Instance&, bool) override {
+    return Http::FilterDataStatus::Continue;
+  }
+  Http::FilterTrailersStatus encodeTrailers(Http::HeaderMap&) override {
+    return Http::FilterTrailersStatus::Continue;
+  }
+  void setEncoderFilterCallbacks(Http::StreamEncoderFilterCallbacks& callbacks) override {
+    encoder_callbacks_ = &callbacks;
+  }
+
+protected:
+  Http::StreamEncoderFilterCallbacks* encoder_callbacks_{};
+};
+
+// A filter which passes all data through with Continue status.
+class PassThroughFilter : public StreamFilter {
```

Comment:
can this just inherit both PassThrough{Decoder,Encoder}Filter ?

Reply:
That's what I wanted to do but I ran into this error: https://ideone.com/d0aImr

Maybe I'm missing something, but I couldn't get it to both satisfy `StreamEncoder` and reuse the implementations in `PassThrough{Decoder,Encoder}Filter`.

Reply:
Oh because StreamFilter doesn't do virtual inheritance like this: https://ideone.com/YEPIBn

Reply:
Awesome, I'll switch to that. I tried playing around with `virtual` but I've never used it much so couldn't quite figure it out